core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.spec.ts

```diff
@@ -69,72 +106,211 @@ export const pagesInitialTestData = [
     });
 
-    describe('Empty State', () => {
-        beforeEach(() => {
-            TestBed.overrideProvider(DotPageStore, {
-                useValue: {
-                    getFavoritePages: jasmine.createSpy(),
-                    setInitialStateData: jasmine.createSpy(),
-                    limitFavoritePages: jasmine.createSpy(),
-                    vm$: of({
-                        favoritePages: {
-                            items: [],
-                            showLoadMoreButton: false,
-                            total: 0
-                        },
-                        loggedUserId: 'admin'
-                    })
-                }
-            });
-            store = TestBed.inject(DotPageStore);
-            fixture = TestBed.createComponent(DotPagesComponent);
-            de = fixture.debugElement;
-            fixture.detectChanges();
+    // describe('Empty State', () => {
+    beforeEach(() => {
+        TestBed.overrideProvider(DotPageStore, {
+            useValue: storeMock
         });
+        store = TestBed.inject(DotPageStore);
+        dotRouterService = TestBed.inject(DotRouterService);
+        dotMessageDisplayService = TestBed.inject(DotMessageDisplayService);
+        fixture = TestBed.createComponent(DotPagesComponent);
+        de = fixture.debugElement;
+        component = fixture.componentInstance;
 
-        it('should init store', () => {
-            expect(store.setInitialStateData).toHaveBeenCalledWith(5);
+        fixture.detectChanges();
+        spyOn(component.menu, 'hide');
+        spyOn(dotMessageDisplayService, 'push');
+    });
+
+    it('should init store', () => {
+        expect(store.setInitialStateData).toHaveBeenCalledWith(5);
+    });
+
+    it('should have favorite page panel, menu, pages panel and DotAddToBundle components', () => {
+        expect(de.query(By.css('dot-pages-favorite-panel'))).toBeTruthy();
+        expect(de.query(By.css('p-menu'))).toBeTruthy();
+        expect(de.query(By.css('dot-pages-listing-panel'))).toBeTruthy();
+        expect(de.query(By.css('dot-add-to-bundle'))).toBeTruthy();
+    });
+
+    it('should call goToUrl method from DotPagesFavoritePanel', () => {
+        const elem = de.query(By.css('dot-pages-favorite-panel'));
+        elem.triggerEventHandler('goToUrl', '/page/1?lang=1');
+
+        expect(dotRouterService.goToEditPage).toHaveBeenCalledWith({
+            lang: '1',
+            url: '/page/1'
         });
+    });
 
+    it('should call showActionsMenu method from DotPagesFavoritePanel', () => {
+        const eventMock = new MouseEvent('click');
+        Object.defineProperty(eventMock, 'currentTarget', {
+            value: { id: 'test' },
+            enumerable: true
+        });
+
+        const actionMenuParam = {
+            event: eventMock,
+            actionMenuDomId: 'test1',
+            item: dotcmsContentletMock
+        };
+
+        const elem = de.query(By.css('dot-pages-favorite-panel'));
```

Comment:
testId.

Reply:
not needed, there will be always only 1 `dot-pages-favorite-panel` component on the page